libs/traits/src/lib.rs

```diff
@@ -629,16 +669,24 @@ pub mod fees {
 }
 
 /// Trait to determine whether a sending account and currency have a
-/// restriction, and if so is there an allowance for the reciever location.
+/// restriction, and if so is there an allowance for the receiver location.
 pub trait TransferAllowance<AccountId> {
 	type CurrencyId;
 	type Location: Member + Debug + Eq + PartialEq + TypeInfo + Encode + Decode + MaxEncodedLen;
 	/// Determines whether the `send` account is allowed to make a transfer to
-	/// the  `receive` location with `currency` type currency. Returns result
+	/// the `receive` location with `currency` type currency. Returns result
 	/// wrapped bool for whether allowance is allowed.
 	fn allowance(
 		send: AccountId,
-		recieve: Self::Location,
+		receive: Self::Location,
 		currency: Self::CurrencyId,
 	) -> DispatchResult;
 }
+
+/// Trait to retrieve information about currencies.
+pub trait CurrencyInspect {
+	type CurrencyId;
+
+	/// Checks whether the provided currency is a tranche token.
+	fn is_tranche_token(currency: Self::CurrencyId) -> bool;
```

Comment:
I am not sure it would be cleaner to move this to `TrancheCurrency` as with the current implementation this is per definition a tranche token. However, we want to check whether an arbitrary currency is in fact a tranche token. Adding such a static check to the trait seems misplaced to me.

Reply:
That's not what I meant, sorry. I meant could just do `TrancheCurrency::generate(_, _) == currency`, which would ensure the input currency is a tranche token currency. But it is clear and clean as is so fine for me to keep it that way

Reply:
Looking over it again, I feel like we can entirely remove this trait. Since `CurrencyId` implements `TrancheCurrency` and we are generating a `TrancheCurrency` before checking `is_tranche_token` anyway, it feels completely redundant.